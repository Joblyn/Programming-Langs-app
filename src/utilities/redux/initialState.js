export const questions = {
  isSuccessful: false,
  isError: false,
  data: null,
  error: null,
};

export const error = {
  isError: false,
  data: null,
};

export const responses = {};
