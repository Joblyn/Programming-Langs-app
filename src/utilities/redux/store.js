import { configureStore } from "@reduxjs/toolkit";
import errorReducer from "./reducers/error";
import { questionsReducer } from "./reducers/questions";
import { responsesReducer } from "./reducers/responses";

export const store = configureStore({
  reducer: {
    questions: questionsReducer,
    responses: responsesReducer,
    connectionError: errorReducer
  }
})