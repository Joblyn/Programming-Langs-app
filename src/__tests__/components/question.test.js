import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Question from "../../components/question/question";
import { Provider } from "react-redux";
import { store } from "../../utilities/redux/store";

describe("<Question />", () => {
  afterEach(cleanup);

  const question = {
    text: "Do yo know Javascript objects?",
    choices: "Yes|No|Maybe",
  };
  // UI test
  it("renders component correctly", () => {
    const { container, getByText } = render(
      <Provider store={store}>
        <Question question={question} />
      </Provider>
    );

    expect(getByText(question.text)).toBeTruthy();
    expect(getByText("Yes")).toBeTruthy();
    expect(getByText("No")).toBeTruthy();
    expect(getByText("Maybe")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
