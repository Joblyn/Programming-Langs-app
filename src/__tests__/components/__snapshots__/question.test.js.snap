// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> renders component correctly 1`] = `
<div
  class="mb-4"
>
  <div
    class="mb-4 flex flex-row items-start"
  >
    <span
      class="text-lg mr-3 mt-1 question_number"
    >
      NaN
      .
    </span>
    <h3
      class="font-bold text-2xl"
    >
      Do yo know Javascript objects?
    </h3>
  </div>
  <div
    class="mt-2 px-6"
  >
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative  opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <span
          class="tracking-wider w-full text-center"
        >
          Yes
        </span>
      </label>
    </div>
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative  opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <span
          class="tracking-wider w-full text-center"
        >
          No
        </span>
      </label>
    </div>
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative  opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <span
          class="tracking-wider w-full text-center"
        >
          Maybe
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`<Question /> renders with defaultValue prop 1`] = `
<div
  class="mb-4"
>
  <div
    class="mb-4 flex flex-row items-start"
  >
    <span
      class="text-lg mr-3 mt-1 question_number"
    >
      NaN
      .
    </span>
    <h3
      class="font-bold text-2xl"
    >
      Do yo know Javascript objects?
    </h3>
  </div>
  <div
    class="mt-2 px-6"
  >
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative checked opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium checked-icon text-white absolute css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41L9 16.17z"
          />
        </svg>
        <span
          class="tracking-wider w-full text-center"
        >
          Yes
        </span>
      </label>
    </div>
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative  opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <span
          class="tracking-wider w-full text-center"
        >
          No
        </span>
      </label>
    </div>
    <div
      class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primary relative  opt-cont active:scale-95"
    >
      <label
        class="w-full p-2.5 opt-label cursor-pointer relative"
      >
        <span
          class="tracking-wider w-full text-center"
        >
          Maybe
        </span>
      </label>
    </div>
  </div>
</div>
`;
